### DISCUSSION TITLE
Can't seem to getCurrentUser() working on a page refresh SSR Nuxt

### DISCUSSION THREAD

**Discussion by Picturk** [vuejs/vuefire#1490] [Questions and Help] [answered]
Attempting to use middleware for route protection. Fails when a user refreshes the webpage

```javascript
definePageMeta({
  name: 'myWebpage',
  middleware: async function (to, from) {
    const currentUser = await getCurrentUser()
    if (currentUser && currentUser.uid === to.params.uid) {
      return true
    } else {
      return navigateTo('/login')
    }
  },
})
```

Above code checks for the presence of a current user via `getCurrentUser()` and validates their access based on a UID match. If the user is not authenticated or their UID does not match the parameter in the route, they are redirected to the login page.

However, the issue arises when an authenticated user refreshes the web page. Instead of maintaining their session on the myWebpage page, they are redirected to the login page (`/login`). 

I am uncertain how to get `getCurrentUser()` working in a server-side context. I running the application on Google Cloud Run. My config is below.

```javascript
...
  vuefire: {
    auth: {
      enabled: true,
      persistence: ['indexedDBLocal'],
    },
    config: {
      apiKey: process.env.FIREBASE_API_KEY,
      authDomain: process.env.FIREBASE_AUTH_DOMAIN,
      projectId: process.env.FIREBASE_PROJECT_ID,
      storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
      messagingSenderId: process.env.FIREBASE_MESSAGING_SENDER_ID,
      appId: process.env.FIREBASE_APP_ID,
      measurementId: process.env.FIREBASE_MEASUREMENT_ID,
    },
    emulators: true,
  },
```
Recommendations on tackling this challenge would be very gratefully received.

**Comment by Picturk** [comment, marked as answer]
solved very simply

```javascript
vuefire: {
    auth: {
      enabled: true,
      persistence: ['indexedDBLocal'],
      sessionCookie: true,
    },
...
```

**Reply by posva** [reply]
Yes: https://vuefire.vuejs.org/nuxt/auth.html#Session-Cookie
